project-icp/bee-pollinator-app
# Only use AWS_PROFILE in development

## Overview

This PR modifies the `bee-pollinator.base` Ansible role to only create `/etc/icp.d/env/AWS_PROFILE` in development. `AWS_PROFILE` should not be set in Production because IAM credentials come from the App Server Instance Profile, not a `boto` config file. This also fixes the AMI build, which was failing because `aws_profile` is not defined in the `packer` environment. Finally, I made sure that the BEEKEEPERS_DATA_BUCKET variable added to the Packer `group_vars`, and the `cloud-config` template for Staging and Production.

Connects #311 

### Demo

Optional. Screenshots, `curl` examples, etc.

### Notes

I've already added the new `BeekeepersDataBucket` input to the `default.yml` CloudFormation input file on the fileshare, and the `icp.yml` CloudFormation input file on Jenkins.

## Testing Instructions

 * This PR has been deployed to Staging. See [Jenkins logs](http://civicci01.internal.azavea.com/job/bee-pollinator-app-and-worker/80/)
 * Hit the `/beekeepers/fetch/` endpoint on Staging, ensure that the App Server can download `.tiff` from the Beekeepers Data Bucket.

```bash
$  curl -k -XPOST https://staging.app.pollinationmapper.org/beekeepers/fetch/ 
{"forage_spring_3k.tif":34.80228042602539,"forage_sum_3k.tif":53.82372283935547,"forage_fall_3k.tif":38.794944763183594,"pesticide_3k.tif":0.26457083225250244}%
```
